crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -1551,23 +1558,52 @@ impl CumulocityConverter {
                 self.config.ops_dir.join(child_dir_name).into()
             }
             EntityType::Service => {
-                let target = &device.topic_id;
-                error!("Unsupported {c8y_operation_name} operation for a service: {target}");
-                return Ok(vec![]);
+                error!(
+                    %target,
+                    "{c8y_operation_name} operation for services are currently unsupported"
+                );
+                return Err(ConversionError::UnexpectedError(anyhow!(
+                    "{c8y_operation_name} operation for services are currently unsupported"
+                )));
             }
         };
+
         let ops_file = ops_dir.join(c8y_operation_name);
         create_directory_with_defaults(&*ops_dir)?;
-        create_file_with_defaults(ops_file, None)?;
+        create_file_with_defaults(&ops_file, None)?;
+
+        let c8y_prefix = self.config.bridge_config.c8y_prefix.clone();
+
+        let operation = c8y_api::smartrest::operations::get_operation(
+            ops_file.as_std_path(),
+            &self.config.bridge_config,
+        )?;
+        let operations = self
+            .operations_for_device_mut(target)
+            .expect("entity should've been checked before that it's not a service");
 
-        let need_cloud_update = self.update_operations(ops_dir.as_std_path())?;
+        let prev_operation = operations.remove_operation(&operation.name);
+        // even if the body of the operation is different, as long as it has the same name, supported operations message
+        // will be the same, so we don't need to resend
+        let need_cloud_update = prev_operation.is_none();
+        operations.add_operation(operation);
 
         if need_cloud_update {
-            let device_operations = self.create_supported_operations(
-                ops_dir.as_std_path(),
-                &self.config.bridge_config.c8y_prefix,
-            )?;
-            return Ok(vec![device_operations]);
+            let topic = if is_child_operation_path(ops_dir.as_std_path()) {
+                let child_id = get_child_external_id(ops_dir.as_std_path())?;
+                let child_external_id = Self::validate_external_id(&child_id)?;
+
+                C8yTopic::ChildSmartRestResponse(child_external_id.into()).to_topic(&c8y_prefix)?
+            } else {
+                C8yTopic::upstream_topic(&c8y_prefix)
+            };
+
+            let cloud_update_operations_message = MqttMessage::new(
+                &topic,
+                operations.create_smartrest_ops_message().into_inner(),
+            );
```

Comment:
Again, why not reuse some of this logic that is already there in the `create_supported_operations` function?

Reply:
Yep, in this case `create_supported_operations` can probably be used and i just missed it.

Reply:
fixed in 01ec205849